### ISSUE TITLE
Remote controller channel change not working

### ISSUE THREAD

**Issue by tomidis** [Me-TV/Me-TV#38]
Up till now I didn't noticed there has been work on the remote controllers support (#17).
For me it's not working the channel change using channel +/- keys.
The volume control using volume +/- keys is working.
I'm using this controller
[http://terratec.ultron.info/Receiver/Cinergy_Hybrid_T_USB_XS/Images/Cinergy_Hybrid_T_USB_XS_remote_control_L.jpg](http://terratec.ultron.info/Receiver/Cinergy_Hybrid_T_USB_XS/Images/Cinergy_Hybrid_T_USB_XS_remote_control_L.jpg)
I tried all available keys, and it's working only volume +/-, and "FR" and "FF" keys.
However, when I press "FR" or "FF" key, I get this:
[https://www.uihere.com/free-icon/cancel-icon-47392](https://www.uihere.com/free-icon/cancel-icon-47392)

**Comment by russel** [comment]
Currently (2019-02-04) only channel up, channel down, volume up and volume down keystrokes are processed. There needs to be a campaign to extend the list of useful, and processed, keys. Until the EPG is working so that Me TV can have knowledge of logical channel numbers, I haven't worried about the numeric keys.

Whilst I have a Terratec Cinergy T xxs, and thus one of those remote controls. I haven't been able to make the DVB device work on my machines for a long while. I'll dig it out again and put some effort into getting it working.

I have relied on the already known to the Linux kernel key lookup tables so far. This has worked for my PCTV and Haupauge adapters. It may be that the Terratec key table does not get loaded automatically. Research required, but get the device working to test is a precursor.

**Comment by russel** [comment]
I just tried the Terratec Cinergy T xxs I have and it installs properly, but it just doesn't play the DVB-T stream at all. Totally weird. However it is the case that it correctly installs a remote control key table:
```
Registered IR keymap rc-dib0700-rc5
```
so it should all just work. That the channel up and channel down do not work is thus strange. I have tried using evtest with my remote and it reports a reasonable table but the controller isn't sending proper signals. I am hypothesising that my device is a bit broken.

**Comment by tomidis** [comment]
Maybe I'm using the wrong rc key table.
I don't know how to install
 > Registered IR keymap rc-dib0700-rc5

I found somewhere and currently using the following procedure to enable the remote control:
in
`/etc/modprobe.d/`
copy the .conf file with the following content:
`options cx23885 enable_885_ir=1`

**Comment by tomidis** [comment]
OK, dmesg output gives me:
`Registered IR keymap rc-nec-terratec-cinergy-xs`

**Comment by russel** [comment]
I'll have to research the "explicitly loading a module" (what you have) route from "letting the kernel choose a module" route (what I have), it may be that the wrong firmware is being chosen in my set up, stopping the device from playing DVB-T signal.

That you have an IR keymap loading is the crucial factor here. What we need to check is that the mapping of the symbols is the same as I am assuming – this potentially depends on Linux versions, but research had indicated that the values haven't changed in years and so should just work.

I use evtest to check things:
```
 |> evtest
No device specified, trying to scan all of /dev/input/event*
Not running as root, no devices may be available.
Available devices:
/dev/input/event0:	AT Translated Set 2 keyboard
/dev/input/event1:	SynPS/2 Synaptics TouchPad
/dev/input/event2:	TPPS/2 IBM TrackPoint
/dev/input/event3:	Lid Switch
/dev/input/event4:	Sleep Button
/dev/input/event5:	Power Button
/dev/input/event6:	PC Speaker
/dev/input/event7:	ThinkPad Extra Buttons
/dev/input/event8:	Integrated Camera: Integrated C
/dev/input/event9:	HDA Digital PCBeep
/dev/input/event10:	HDA Intel PCH Dock Mic
/dev/input/event11:	HDA Intel PCH Mic
/dev/input/event12:	HDA Intel PCH Dock Headphone
/dev/input/event13:	HDA Intel PCH Headphone
/dev/input/event14:	Video Bus
/dev/input/event15:	HDA Intel HDMI HDMI/DP,pcm=3
/dev/input/event16:	HDA Intel HDMI HDMI/DP,pcm=7
/dev/input/event17:	HDA Intel HDMI HDMI/DP,pcm=8
/dev/input/event18:	HDA Intel HDMI HDMI/DP,pcm=9
/dev/input/event19:	HDA Intel HDMI HDMI/DP,pcm=10
/dev/input/event20:	Logitech Wireless Mouse
/dev/input/event21:	Terratec Cinergy T USB XXS (HD)/ T3
Select the device event number [0-21]: 21
Input driver version is 1.0.1
Input device ID: bus 0x3 vendor 0xccd product 0xab version 0x100
Input device name: "Terratec Cinergy T USB XXS (HD)/ T3"
Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 2 (KEY_1)
    Event code 3 (KEY_2)
    Event code 4 (KEY_3)
    Event code 5 (KEY_4)
    Event code 6 (KEY_5)
    Event code 7 (KEY_6)
    Event code 8 (KEY_7)
    Event code 9 (KEY_8)
    Event code 10 (KEY_9)
    Event code 11 (KEY_0)
    Event code 41 (KEY_GRAVE)
    Event code 52 (KEY_DOT)
    Event code 55 (KEY_KPASTERISK)
    Event code 102 (KEY_HOME)
    Event code 103 (KEY_UP)
    Event code 105 (KEY_LEFT)
    Event code 106 (KEY_RIGHT)
    Event code 108 (KEY_DOWN)
    Event code 111 (KEY_DELETE)
    Event code 113 (KEY_MUTE)
    Event code 114 (KEY_VOLUMEDOWN)
    Event code 115 (KEY_VOLUMEUP)
    Event code 116 (KEY_POWER)
    Event code 119 (KEY_PAUSE)
    Event code 128 (KEY_STOP)
    Event code 139 (KEY_MENU)
    Event code 158 (KEY_BACK)
    Event code 164 (KEY_PLAYPAUSE)
    Event code 167 (KEY_RECORD)
    Event code 168 (KEY_REWIND)
    Event code 173 (KEY_REFRESH)
    Event code 207 (KEY_PLAY)
    Event code 208 (KEY_FASTFORWARD)
    Event code 210 (KEY_PRINT)
    Event code 223 (KEY_CANCEL)
    Event code 226 (KEY_MEDIA)
    Event code 352 (KEY_OK)
    Event code 354 (KEY_GOTO)
    Event code 355 (KEY_CLEAR)
    Event code 358 (KEY_INFO)
    Event code 363 (KEY_CHANNEL)
    Event code 365 (KEY_EPG)
    Event code 370 (KEY_SUBTITLE)
    Event code 375 (KEY_SCREEN)
    Event code 377 (KEY_TV)
    Event code 385 (KEY_RADIO)
    Event code 386 (KEY_TUNER)
    Event code 388 (KEY_TEXT)
    Event code 389 (KEY_DVD)
    Event code 392 (KEY_AUDIO)
    Event code 393 (KEY_VIDEO)
    Event code 398 (KEY_RED)
    Event code 399 (KEY_GREEN)
    Event code 400 (KEY_YELLOW)
    Event code 401 (KEY_BLUE)
    Event code 402 (KEY_CHANNELUP)
    Event code 403 (KEY_CHANNELDOWN)
    Event code 405 (KEY_LAST)
    Event code 407 (KEY_NEXT)
    Event code 410 (KEY_SHUFFLE)
    Event code 412 (KEY_PREVIOUS)
  Event type 4 (EV_MSC)
    Event code 4 (MSC_SCAN)
Key repeat handling:
  Repeat type 20 (EV_REP)
    Repeat code 0 (REP_DELAY)
      Value    500
    Repeat code 1 (REP_PERIOD)
      Value    125
Properties:
Testing ... (interrupt to exit)

```

**Comment by tomidis** [comment]
There are some differences, but KEY_CHANNELUP and KEY_CHANNELDOWN are the same
```
|> evtest
No device specified, trying to scan all of /dev/input/event*
Available devices:
/dev/input/event0:	Sleep Button
/dev/input/event1:	Power Button
/dev/input/event2:	Logitech USB Optical Mouse
/dev/input/event3:	Logitech HID compliant keyboard
/dev/input/event4:	Logitech HID compliant keyboard
/dev/input/event5:	HDA ATI HDMI HDMI/DP,pcm=3
/dev/input/event6:	cx23885 IR (TerraTec Cinergy T PCIe Dual)
/dev/input/event7:	Lakewest Audio Audiolab M-DAC
Select the device event number [0-7]: 6
Input driver version is 1.0.1
Input device ID: bus 0x1 vendor 0x153b product 0x117e version 0x1
Input device name: "cx23885 IR (TerraTec Cinergy T PCIe Dual)"
Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 2 (KEY_1)
    Event code 3 (KEY_2)
    Event code 4 (KEY_3)
    Event code 5 (KEY_4)
    Event code 6 (KEY_5)
    Event code 7 (KEY_6)
    Event code 8 (KEY_7)
    Event code 9 (KEY_8)
    Event code 10 (KEY_9)
    Event code 11 (KEY_0)
    Event code 14 (KEY_BACKSPACE)
    Event code 102 (KEY_HOME)
    Event code 103 (KEY_UP)
    Event code 105 (KEY_LEFT)
    Event code 106 (KEY_RIGHT)
    Event code 108 (KEY_DOWN)
    Event code 111 (KEY_DELETE)
    Event code 113 (KEY_MUTE)
    Event code 114 (KEY_VOLUMEDOWN)
    Event code 115 (KEY_VOLUMEUP)
    Event code 119 (KEY_PAUSE)
    Event code 128 (KEY_STOP)
    Event code 139 (KEY_MENU)
    Event code 167 (KEY_RECORD)
    Event code 168 (KEY_REWIND)
    Event code 173 (KEY_REFRESH)
    Event code 207 (KEY_PLAY)
    Event code 208 (KEY_FASTFORWARD)
    Event code 212 (KEY_CAMERA)
    Event code 352 (KEY_OK)
    Event code 356 (KEY_POWER2)
    Event code 358 (KEY_INFO)
    Event code 365 (KEY_EPG)
    Event code 370 (KEY_SUBTITLE)
    Event code 373 (KEY_MODE)
    Event code 377 (KEY_TV)
    Event code 385 (KEY_RADIO)
    Event code 386 (KEY_TUNER)
    Event code 388 (KEY_TEXT)
    Event code 389 (KEY_DVD)
    Event code 393 (KEY_VIDEO)
    Event code 398 (KEY_RED)
    Event code 399 (KEY_GREEN)
    Event code 400 (KEY_YELLOW)
    Event code 401 (KEY_BLUE)
    Event code 402 (KEY_CHANNELUP)
    Event code 403 (KEY_CHANNELDOWN)
    Event code 405 (KEY_LAST)
    Event code 407 (KEY_NEXT)
  Event type 4 (EV_MSC)
    Event code 4 (MSC_SCAN)
Key repeat handling:
  Repeat type 20 (EV_REP)
    Repeat code 0 (REP_DELAY)
      Value    500
    Repeat code 1 (REP_PERIOD)
      Value    125
Properties:
Testing ... (interrupt to exit)

```

**Comment by russel** [comment]
As long as the key codes are the same for each "keystroke" we are interested in, things should Just Work™ . So in this case, the hypothesis is that KEY_VOLUMEUP, KEY_VOLUMEDOWN, KEY_CHANNELUP, and KEY_CHANNELDOWN should work with Me TV, though no others will for the moment anyway.

The next test using evtest is to actually try the buttons. Using an adapter I know works, rather than perservering with my Terratec which appears not to:
```
|> evtest
No device specified, trying to scan all of /dev/input/event*
Not running as root, no devices may be available.
Available devices:
/dev/input/event0:	Power Button
/dev/input/event1:	Power Button
/dev/input/event2:	PC Speaker
/dev/input/event3:	Xonar DSX Headphone
/dev/input/event4:	HDA Digital PCBeep
/dev/input/event5:	HDA Intel Mic
/dev/input/event6:	HDA Intel Line
/dev/input/event7:	HDA Intel Line Out
/dev/input/event8:	HDA Intel Front Headphone
/dev/input/event9:	HDA NVidia HDMI/DP,pcm=3
/dev/input/event10:	HDA NVidia HDMI/DP,pcm=7
/dev/input/event11:	Telink TeckNet X315 Mouse
/dev/input/event12:	Telink TeckNet X315 Consumer Control
/dev/input/event13:	Telink TeckNet X315 System Control
/dev/input/event14:	Telink TeckNet X315
/dev/input/event15:	Logitech M525
/dev/input/event16:	1-5.2.4.1.2:1.0 IR
Select the device event number [0-16]: 16
Input driver version is 1.0.1
Input device ID: bus 0x3 vendor 0x2040 product 0x264 version 0x1
Input device name: "1-5.2.4.1.2:1.0 IR"
Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 2 (KEY_1)
    Event code 3 (KEY_2)
    Event code 4 (KEY_3)
    Event code 5 (KEY_4)
    Event code 6 (KEY_5)
    Event code 7 (KEY_6)
    Event code 8 (KEY_7)
    Event code 9 (KEY_8)
    Event code 10 (KEY_9)
    Event code 11 (KEY_0)
    Event code 113 (KEY_MUTE)
    Event code 114 (KEY_VOLUMEDOWN)
    Event code 115 (KEY_VOLUMEUP)
    Event code 116 (KEY_POWER)
    Event code 128 (KEY_STOP)
    Event code 138 (KEY_HELP)
    Event code 139 (KEY_MENU)
    Event code 164 (KEY_PLAYPAUSE)
    Event code 167 (KEY_RECORD)
    Event code 168 (KEY_REWIND)
    Event code 208 (KEY_FASTFORWARD)
    Event code 352 (KEY_OK)
    Event code 370 (KEY_SUBTITLE)
    Event code 372 (KEY_ZOOM)
    Event code 402 (KEY_CHANNELUP)
    Event code 403 (KEY_CHANNELDOWN)
  Event type 4 (EV_MSC)
    Event code 4 (MSC_SCAN)
Key repeat handling:
  Repeat type 20 (EV_REP)
    Repeat code 0 (REP_DELAY)
      Value    500
    Repeat code 1 (REP_PERIOD)
      Value    125
Properties:
Testing ... (interrupt to exit)
Event: time 1549432431.649793, type 4 (EV_MSC), code 4 (MSC_SCAN), value 703
Event: time 1549432431.649793, type 1 (EV_KEY), code 115 (KEY_VOLUMEUP), value 1
Event: time 1549432431.649793, -------------- SYN_REPORT ------------
Event: time 1549432431.857789, type 4 (EV_MSC), code 4 (MSC_SCAN), value 703
Event: time 1549432431.857789, -------------- SYN_REPORT ------------
Event: time 1549432432.101709, type 1 (EV_KEY), code 115 (KEY_VOLUMEUP), value 0
Event: time 1549432432.101709, -------------- SYN_REPORT ------------
Event: time 1549432437.057774, type 4 (EV_MSC), code 4 (MSC_SCAN), value 703
Event: time 1549432437.057774, type 1 (EV_KEY), code 115 (KEY_VOLUMEUP), value 1
Event: time 1549432437.057774, -------------- SYN_REPORT ------------
Event: time 1549432437.265765, type 4 (EV_MSC), code 4 (MSC_SCAN), value 703
Event: time 1549432437.265765, -------------- SYN_REPORT ------------
Event: time 1549432437.509708, type 1 (EV_KEY), code 115 (KEY_VOLUMEUP), value 0
Event: time 1549432437.509708, -------------- SYN_REPORT ------------
Event: time 1549432438.737766, type 4 (EV_MSC), code 4 (MSC_SCAN), value 709
Event: time 1549432438.737766, type 1 (EV_KEY), code 114 (KEY_VOLUMEDOWN), value 1
Event: time 1549432438.737766, -------------- SYN_REPORT ------------
Event: time 1549432438.945764, type 4 (EV_MSC), code 4 (MSC_SCAN), value 709
Event: time 1549432438.945764, -------------- SYN_REPORT ------------
Event: time 1549432439.189712, type 1 (EV_KEY), code 114 (KEY_VOLUMEDOWN), value 0
Event: time 1549432439.189712, -------------- SYN_REPORT ------------
Event: time 1549432441.441756, type 4 (EV_MSC), code 4 (MSC_SCAN), value 709
Event: time 1549432441.441756, type 1 (EV_KEY), code 114 (KEY_VOLUMEDOWN), value 1
Event: time 1549432441.441756, -------------- SYN_REPORT ------------
Event: time 1549432441.649879, type 4 (EV_MSC), code 4 (MSC_SCAN), value 709
Event: time 1549432441.649879, -------------- SYN_REPORT ------------
Event: time 1549432441.893709, type 1 (EV_KEY), code 114 (KEY_VOLUMEDOWN), value 0
Event: time 1549432441.893709, -------------- SYN_REPORT ------------
Event: time 1549432444.457868, type 4 (EV_MSC), code 4 (MSC_SCAN), value 70c
Event: time 1549432444.457868, type 1 (EV_KEY), code 403 (KEY_CHANNELDOWN), value 1
Event: time 1549432444.457868, -------------- SYN_REPORT ------------
Event: time 1549432444.665862, type 4 (EV_MSC), code 4 (MSC_SCAN), value 70c
Event: time 1549432444.665862, -------------- SYN_REPORT ------------
Event: time 1549432444.909706, type 1 (EV_KEY), code 403 (KEY_CHANNELDOWN), value 0
Event: time 1549432444.909706, -------------- SYN_REPORT ------------
Event: time 1549432446.121858, type 4 (EV_MSC), code 4 (MSC_SCAN), value 706
Event: time 1549432446.121858, type 1 (EV_KEY), code 402 (KEY_CHANNELUP), value 1
Event: time 1549432446.121858, -------------- SYN_REPORT ------------
Event: time 1549432446.329854, type 4 (EV_MSC), code 4 (MSC_SCAN), value 706
Event: time 1549432446.329854, -------------- SYN_REPORT ------------
Event: time 1549432446.573705, type 1 (EV_KEY), code 402 (KEY_CHANNELUP), value 0
Event: time 1549432446.573705, -------------- SYN_REPORT ------------
Event: time 1549432447.473854, type 4 (EV_MSC), code 4 (MSC_SCAN), value 706
Event: time 1549432447.473854, type 1 (EV_KEY), code 402 (KEY_CHANNELUP), value 1
Event: time 1549432447.473854, -------------- SYN_REPORT ------------
Event: time 1549432447.681850, type 4 (EV_MSC), code 4 (MSC_SCAN), value 706
Event: time 1549432447.681850, -------------- SYN_REPORT ------------
Event: time 1549432447.925707, type 1 (EV_KEY), code 402 (KEY_CHANNELUP), value 0
Event: time 1549432447.925707, -------------- SYN_REPORT ------------

```

**Comment by tomidis** [comment]
It seems that the keys are working. However, I noticed that remote control volume keys are working no matter the Me-TV is running or not. Using the volume keys, I'm affecting all system and applications sounds. Is this normal or the remote control keys must change only the Me-TV's volume (when Me-TV is running). If this is true, then my Me-TV doesn't respond to either volume or channel keys. 
```
|> evtest
No device specified, trying to scan all of /dev/input/event*
Available devices:
/dev/input/event0:	Sleep Button
/dev/input/event1:	Power Button
/dev/input/event2:	Lakewest Audio Audiolab M-DAC
/dev/input/event3:	Logitech USB Optical Mouse
/dev/input/event4:	Logitech HID compliant keyboard
/dev/input/event5:	Logitech HID compliant keyboard
/dev/input/event6:	HDA ATI HDMI HDMI/DP,pcm=3
/dev/input/event7:	cx23885 IR (TerraTec Cinergy T PCIe Dual)
Select the device event number [0-7]: 7
Input driver version is 1.0.1
Input device ID: bus 0x1 vendor 0x153b product 0x117e version 0x1
Input device name: "cx23885 IR (TerraTec Cinergy T PCIe Dual)"
Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 2 (KEY_1)
    Event code 3 (KEY_2)
    Event code 4 (KEY_3)
    Event code 5 (KEY_4)
    Event code 6 (KEY_5)
    Event code 7 (KEY_6)
    Event code 8 (KEY_7)
    Event code 9 (KEY_8)
    Event code 10 (KEY_9)
    Event code 11 (KEY_0)
    Event code 14 (KEY_BACKSPACE)
    Event code 102 (KEY_HOME)
    Event code 103 (KEY_UP)
    Event code 105 (KEY_LEFT)
    Event code 106 (KEY_RIGHT)
    Event code 108 (KEY_DOWN)
    Event code 111 (KEY_DELETE)
    Event code 113 (KEY_MUTE)
    Event code 114 (KEY_VOLUMEDOWN)
    Event code 115 (KEY_VOLUMEUP)
    Event code 119 (KEY_PAUSE)
    Event code 128 (KEY_STOP)
    Event code 139 (KEY_MENU)
    Event code 167 (KEY_RECORD)
    Event code 168 (KEY_REWIND)
    Event code 173 (KEY_REFRESH)
    Event code 207 (KEY_PLAY)
    Event code 208 (KEY_FASTFORWARD)
    Event code 212 (KEY_CAMERA)
    Event code 352 (KEY_OK)
    Event code 356 (KEY_POWER2)
    Event code 358 (KEY_INFO)
    Event code 365 (KEY_EPG)
    Event code 370 (KEY_SUBTITLE)
    Event code 373 (KEY_MODE)
    Event code 377 (KEY_TV)
    Event code 385 (KEY_RADIO)
    Event code 386 (KEY_TUNER)
    Event code 388 (KEY_TEXT)
    Event code 389 (KEY_DVD)
    Event code 393 (KEY_VIDEO)
    Event code 398 (KEY_RED)
    Event code 399 (KEY_GREEN)
    Event code 400 (KEY_YELLOW)
    Event code 401 (KEY_BLUE)
    Event code 402 (KEY_CHANNELUP)
    Event code 403 (KEY_CHANNELDOWN)
    Event code 405 (KEY_LAST)
    Event code 407 (KEY_NEXT)
  Event type 4 (EV_MSC)
    Event code 4 (MSC_SCAN)
Key repeat handling:
  Repeat type 20 (EV_REP)
    Repeat code 0 (REP_DELAY)
      Value    500
    Repeat code 1 (REP_PERIOD)
      Value    125
Properties:
Testing ... (interrupt to exit)
Event: time 1549466731.857165, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141c
Event: time 1549466731.857165, type 1 (EV_KEY), code 115 (KEY_VOLUMEUP), value 1
Event: time 1549466731.857165, -------------- SYN_REPORT ------------
Event: time 1549466731.909540, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141c
Event: time 1549466731.909540, -------------- SYN_REPORT ------------
Event: time 1549466732.164334, type 1 (EV_KEY), code 115 (KEY_VOLUMEUP), value 0
Event: time 1549466732.164334, -------------- SYN_REPORT ------------
Event: time 1549466736.964671, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141e
Event: time 1549466736.964671, type 1 (EV_KEY), code 114 (KEY_VOLUMEDOWN), value 1
Event: time 1549466736.964671, -------------- SYN_REPORT ------------
Event: time 1549466737.016640, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141e
Event: time 1549466737.016640, -------------- SYN_REPORT ------------
Event: time 1549466737.280365, type 1 (EV_KEY), code 114 (KEY_VOLUMEDOWN), value 0
Event: time 1549466737.280365, -------------- SYN_REPORT ------------
Event: time 1549466740.764975, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141b
Event: time 1549466740.764975, type 1 (EV_KEY), code 402 (KEY_CHANNELUP), value 1
Event: time 1549466740.764975, -------------- SYN_REPORT ------------
Event: time 1549466740.817456, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141b
Event: time 1549466740.817456, -------------- SYN_REPORT ------------
Event: time 1549466740.925490, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141b
Event: time 1549466740.925490, -------------- SYN_REPORT ------------
Event: time 1549466741.188421, type 1 (EV_KEY), code 402 (KEY_CHANNELUP), value 0
Event: time 1549466741.188421, -------------- SYN_REPORT ------------
Event: time 1549466743.828972, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141f
Event: time 1549466743.828972, type 1 (EV_KEY), code 403 (KEY_CHANNELDOWN), value 1
Event: time 1549466743.828972, -------------- SYN_REPORT ------------
Event: time 1549466743.881563, type 4 (EV_MSC), code 4 (MSC_SCAN), value 141f
Event: time 1549466743.881563, -------------- SYN_REPORT ------------
Event: time 1549466744.160362, type 1 (EV_KEY), code 403 (KEY_CHANNELDOWN), value 0
Event: time 1549466744.160362, -------------- SYN_REPORT ------------
```

**Comment by russel** [comment]
If the DVB adapter is in place but Me TV is not running, the keystrokes will be picked up by any and all listeners. Once Me TV is running, it takes control of all keystrokes from the DVB adapters, the other listeners do not get the events.

**Comment by tomidis** [comment]
> Once Me TV is running, it takes control of all keystrokes from the DVB adapters, the other listeners do not get the events.

But then my Me TV isn't responding to either volume or channel keys. Because when the Me TV is running, changing the volume by remote controller isn't only for Me TV, but for the whole system sounds (the same like the Me TV not running).

**Comment by russel** [comment]
Hummm… that implies Me TV is not picking up the remote (and grabbing sole access) even though it is clearly there.

I believe I am assuming that your dual tuner PCIe card is setting up /dev/lirc0 as the LIRC device as happens with the USB devices. Me TV looks for /dev/lircX as the start point to work out what remote controls are present.

**Comment by tomidis** [comment]
I didn't have lirc installed, so I installed it and this completely screwed up the use of  remote controller (it seems like after the first pressing of e.g. volume key, it constantly holds this event).
Then I used Timeshift to return the system to previous working state.
I don't know how to check /dev/lirc0 setting?

**Comment by russel** [comment]
My apologies, I didn't give enough data. Whilst there are LIRC packages and a whole server, much of the functionality got moved into the kernel – so for recent kernels there is no need of the LIRC packages for LIRC to work. When a kernel recognised adapter is present, the file /dev/lirc0 (for the first) gets created. It sounds as though your PCIe card is creating a remote control device, but that it isn't /dev/lirc0.

**Comment by tomidis** [comment]
OK, so lirc0 is a file located in /dev
Yes, I have lirc0 file in this location
I created lircx, but after system restart it's gone
Can this be changed in Me TV code, so it searches for any of these files?

@russel 
you are forcing me to learn linux :-)

**Comment by russel** [comment]
The kernel creates /dev/lircX files when it detects a device with a remote control – all files in /dev are actually not real files per se they are file-like objects for communicating between applications and the kernel. So any manually created files will get removed on reboot.

Me TV searches for /dev/lircX on start up, and then sets off a listener in case any new ones get created or extant ones removed. When Me TV finds a /dev/lircX device associates with a DVB adaptor, it grabs sole access to the device so as to stop it controlling the machine at large.

That things are not working for you as they are for me, indicates there is an assumption somewhere in the code that works for me, but not for you.

**Comment by tomidis** [comment]
I guess this issue is solved in v3.0.9
Now I can change the channels with arrow keys at RC, but I'm not sure is this working properly:
while volume change works immediately after pressing volume +/-  keys, the channels change working after a few presses of the arrow key. The first press of the key shows the menu bar (in full screen mode), the second press changes the focus to channels bar, and finally the third press changes the channel. Sometimes it works only after two pressing. And there is also a situation where it working at single press, but this works only within 5 seconds of the menu bar visibility, while the focus is still present on the channels bar, from the previous channel change.
I noticed also that pressing of any key shows the menu bar.

**Comment by russel** [comment]
This is weird. For me, with a WinTV-dualHD and Debian Sid using GStreamer 1.16.0, the channel changes with each click. The problem I have is that not all stations can be tuned to quickly so there is often a long delay between changing channel and the video/audio stream actually playing, and sometimes a video/audio stream cannot be created because the signal is too weak. I do not get the control bar being shown with a channel change click.

**Comment by tomidis** [comment]
Can it be because of the GStreamer version?
I have version 1.14.1
How can I update to 1.16.0 ?
When trying to update
``sudo apt-get install libgstreamer1.0-0``
I get
``libgstreamer1.0-0 is already the newest version (1.14.1-1~ubuntu18.04.1)``

**Comment by dougK44883** [comment]
debian package search shows [libgstreamer1.0-0](https://packages.debian.org/search?suite=default&section=all&arch=any&searchon=names&keywords=libgstreamer1.0-0) only goes to 1.14.4-1 in sid (unstable), you have to use experimental to get 1.16.0-1

I don't use a remote, so can offer no practical input on this topic

**Comment by russel** [comment]
Debian is in Buster release freeze so the new GStreamer release cannot make it to Sid, it can only be in Experimental. At least for now. GStreamer 1.14.4 is thus the official GStreamer for Debian Buster. As soon as the release freeze is over GStreamer 1.16.0 will appear in Sid, and thus hopefully in the next Ubuntu and thus Mint.

Me TV 3.0.9, and indeed the whole 3.0.X series should work fine with GStreamer 1.14.4.

The reason for me using GStreamer 1.16.0 from Experimental is that I want to start on the Me TV 3.1.0 release which will require GStreamer 1.16.0 so as to be able to use the MPEG-TS library and thus process the EPG, and other things.

**Comment by russel** [comment]
It's been a while, sorry. Long story.

Is this still an issue?

**Comment by tomidis** [comment]
Yes, still the same. Need few clicks to change the channel.
I'm currently using Me TV 3.0.9, GStreamer 1.14.5, rustc 1.36.0, Mint 19.1, Cinnamon 4.0.10, Linux 4.15.0-70
I just tried to build a fresh MeTV, but it requested GStreamer 1.16
Can you help me how to install GStreamer 1.16 ?

**Comment by russel** [comment]
As far as I can tell Mint is always based on an Ubuntu LTS distribution. This implies GStreamer 1.16 will only be available once Ubuntu 20.04 is released. GStreamer 1.16 is already in Ubuntu Eoan (19.10) and Focal (will be 20.04 LTS?) though, just as it is in Debian Sid. I know little of Mint so have no idea if it is possible to install GStreamer 1.16 via packages – installing manually leads to a whole mess of stuff that end users really shouldn't have to deal with.

**Comment by russel** [comment]
As far as I can see, as at 2020-04-02, channel changing using a remote fails to work. This has become serious!

**Comment by russel** [comment]
@tomidis  It all seems to work now for me. I have just released v3.0.10, executables available on Bintray (https://bintray.com/beta/#/me-tv/Downloads/Me-TV/v3.0.10?tab=files). If there are problems I am up for fixing things and releasing v3.0.11.

**Comment by tomidis** [comment]
v3.0.10 executable not working
I guess it's because of the GStreamer version. I'm still on Mint 19 and don't have v1.16
Mint 20 is a month away, so then we'll see..

**Comment by russel** [comment]
The executables should work with GStreamer 1.14. That they do not would be useful information.

I will try and create Flatpak distributions to stop all this hassle.

**Comment by russel** [comment]
On the grounds "it works for me", i.e. with my Hauppauge! WinTV-dualHD with Me TV v3.0.10 and master, the volume up and down, and channel up and down work as expected, I am going to close this issue. If it is still a problem for others, then either reopening this issue or opening a new issue will start investigations.
